drivers/ethernet/eth_sam_gmac.c

```diff
@@ -1306,6 +1332,12 @@ static int eth_tx(struct device *dev, struct net_pkt *pkt)
 		__ASSERT(tx_desc_list->head != tx_desc_list->tail,
 			 "tx_desc_list overflow");
 
+		/* Account for a sent frag */
+		ring_buf_put(&queue->tx_frag_list, POINTER_TO_UINT(frag));
+
+		/* frag is internally queued, so it requires to hold a reference */
+		net_buf_ref(frag);
```

Comment:
net_pkt_frag_ref() (it helps to debug when debugging net_pkt buffer allocation)

Reply:
> I am not entirely sure why you keep a reference on all net_pkt's net_buf. In case of PR #12563 being applied, only the very first would be needed.

There are a few reasons for that:
- Given there is no guarantee that the net_pkt is unmutable, I don't want to assume that only the first one is changed. It's true with PR #12563, but that's something that can change with another commit and break.
- It's simpler in the driver to handle all net_buf the same way.
- It allows to stop referencing the net_pkt when PTP is not used.

> net_pkt_frag_ref() (it helps to debug when debugging net_pkt buffer allocation)

Ok, I'll fix that.

Reply:
Another reason: it's symmetric with the RX queue.